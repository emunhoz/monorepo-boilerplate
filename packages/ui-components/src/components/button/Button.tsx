import React from 'react'
import * as S from './Button.style'

interface IButton {
  children?: React.ReactNode
  onClick?: () => void
  disabled?: boolean
  /**
   * Styles button as 100% width
   *
   * @default false
   */
  block?: boolean
  /**
   * Defines the size of the button
   *  big: 66px
   *  small: 32px
   *
   * @default default
   */
  size?: 'big' | 'default' | 'small'

  /**
   * Styles button as negative style
   *
   * @default false
   */
  negative?: boolean

  /**
   * Styles button as ghost style
   *
   * @default false
   */
  ghost?: boolean
}

export const Button = ({ children, disabled, size = 'default', onClick, block = false, negative = false, ghost = false, ...rest }: IButton) => (
  <S.Button onClick={onClick} block={block} size={size} disabled={disabled} negative={negative} ghost={ghost} {...rest}>
    {children}
  </S.Button>
)
