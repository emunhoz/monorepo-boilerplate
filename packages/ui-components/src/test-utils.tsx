import { render } from '@testing-library/react'
import { CustomThemeProvider } from './index'

const AllTheProviders = ({ children }: any) => <CustomThemeProvider>{children}</CustomThemeProvider>

const customRender = (ui: any, options?: any) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
