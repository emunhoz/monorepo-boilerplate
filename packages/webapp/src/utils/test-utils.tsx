import createTheme from '@monorepo/design-tokens'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: any) => (
  <ThemeProvider theme={createTheme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
)

const customRender = (ui: any, options?: any) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
