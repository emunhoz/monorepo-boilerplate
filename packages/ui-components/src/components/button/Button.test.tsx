import React from 'react'
import { render, fireEvent, screen } from '../../utils/test-utils'
import { Button } from './Button'

describe('[Component]: Button', () => {
  const mockOnClickFunction = jest.fn()

  test('render the component', () => {
    const { asFragment } = render(<Button onClick={mockOnClickFunction}>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  test('render negative style', () => {
    const { asFragment } = render(
      <Button negative onClick={mockOnClickFunction}>
        Hello world
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('render ghost style', () => {
    const { asFragment } = render(
      <Button ghost onClick={mockOnClickFunction}>
        Hello world
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('render component with small size style', () => {
    const { asFragment } = render(
      <Button size="small" onClick={mockOnClickFunction}>
        Hello world
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('render component with big size style', () => {
    const { asFragment } = render(
      <Button size="big" onClick={mockOnClickFunction}>
        Hello world
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger function when button is pressed', () => {
    render(<Button onClick={mockOnClickFunction}>Hello world</Button>)

    fireEvent.click(screen.getByText('Hello world'))

    expect(mockOnClickFunction).toHaveBeenCalled()
  })

  test('render component disabled', () => {
    const { asFragment } = render(
      <Button disabled onClick={mockOnClickFunction}>
        Hello world
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
