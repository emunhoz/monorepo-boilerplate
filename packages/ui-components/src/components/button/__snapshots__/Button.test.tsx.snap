// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Button render component disabled 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL iFgXmE"
    disabled=""
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with big size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL hQKOyZ"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with small size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL fdepNw"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render ghost style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL itaVFs"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render negative style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL kALWDj"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render the component 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL iFgXmE"
  >
    Hello world
  </button>
</DocumentFragment>
`;
